llvm/lib/Target/SPIRV/SPIRVModuleAnalysis.cpp

```diff
@@ -967,11 +1117,16 @@ void addInstrRequirements(const MachineInstr &MI,
   case SPIRV::OpConstantSampler:
     Reqs.addCapability(SPIRV::Capability::LiteralSampler);
     break;
+  case SPIRV::OpAccessChain:
+    addOpAccessChainReqs(MI, Reqs, ST);
+    break;
   case SPIRV::OpTypeImage:
     addOpTypeImageReqs(MI, Reqs, ST);
     break;
   case SPIRV::OpTypeSampler:
-    Reqs.addCapability(SPIRV::Capability::ImageBasic);
+    if (ST.isOpenCLEnv()) {
```

Comment:
It's a subtle difference, but recently there were so many talks about "computational flavor of SPIRV but not OpenCL" that I'd probably use rather `!isVulkanEnv()` than `isOpenCLEnv()`. Please feel to ignore though, because at the moment we have just those 2 checks and nobody hasn't yet mentioned enhancing `isOpenCLEnv()` to OpenCL and Non OpenCL.

Reply:
Changed.